storage/innobase/handler/handler0alter.cc

```diff
@@ -5953,6 +5978,242 @@ static bool innobase_instant_try(
 	return false;
 }
 
+/** Update metadata BLOB to reflect updated persistent count.
+@param[in]  user_table  InnoDB table
+@param[in]	table		MySQL table
+@param[in,out]	trx		dictionary transaction
+@retval	true	failure
+@retval	false	success */
+bool innobase_update_persistent_count(
+	dict_table_t* user_table,
+	const TABLE*  table,
+	trx_t*		  trx)
+{
+	if (user_table->alter_persistent_count) {
+		/* ALTER TABLE {tbl} ENABLE|DISABLE PERSISTENT_COUNT is underway */
+		return true;
+	}
+    
+	mem_heap_t* heap = mem_heap_create(1024);
+	dict_index_t* index = dict_table_get_first_index(user_table);
+	mtr_t mtr;
+	mtr.start();
+	/* Prevent purge from calling dict_index_t::clear_instant_alter(),
+	to protect index->n_core_fields, index->table->instant and others
+	from changing during instant_column(). */
+	instant_metadata_lock(*index, mtr);
+	const unsigned n_old_fields = index->n_fields;
+
+	mutex_enter(&dict_sys.mutex);
```

Comment:
Could this be violating the latching order? Did you test with `./mtr --mysqld=--loose-innodb-sync-debug`?